### PR TITLE
fix(button): provides context for AT users when used as reference element for collapsible content

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #6033 

## Summary

Provides context of expanded for AT users when `calcite-button` is used as reference element for collapsible content.

### CODE REVIEW

**Review by driskull — APPROVED** [review]
👍 

I wonder if this deserves its own `expanded` prop? Otherwise users would never know about setting aria-expanded and having it work as expected.

**Comment by anveshmekala** [comment]
> I wonder if this deserves its own `expanded` prop? Otherwise users would never know about setting aria-expanded and having it work as expected.

Using prop seems like a good option. we can explore having an prop which needs to be set by the user when they use button as trigger for any floating/expandable UI. Defining this helps a lot of components which can be used as trigger elements. (ex: fab , action) . It will be additional work for user but they will have the benifit of consistent screen reader experience. 

Thoughts @jcfranco

**Comment by jcfranco** [comment]
For the time being, I think we should skip the prop for a few reasons:

* API concerns
  * we already have `expanded` in some cases where this might fit nicely, but it doesn't translate as well w/ `open`, which is the preferred prop name.
  * we would have to add matching props for other, if not all, attributes.
* This follows existing a11y practices.
* At this point, this is mostly needed for triggers (reference elements). We could revisit this later as components are used in more a11y patterns.

File: packages/calcite-components/src/components/button/button.e2e.ts [hunk 1]
```diff
@@ -644,4 +644,27 @@ describe("calcite-button", () => {
     expect(button1.textContent.length).toBeLessThan(longText.length);
     expect(button1.getAttribute("title")).toEqual(longText);
   });
+
+  it("should set aria-expanded attribute on shadowDOM element when used as trigger", async () => {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/button/button.e2e.ts:648]: Could you test if the `accessible` helper picks this up (also checking that it fails when the ARIA attribute isn't set on the internal button)? If so, I'd suggest using that to simplify this test.

File: packages/calcite-components/src/utils/globalAttributes.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import { createObserver } from "./observers";
 
 type AttributeObject = { [k: string]: any };
-type AllowedGlobalAttribute = "lang" | "role";
-const allowedGlobalAttributes = ["lang", "role"];
+type AllowedGlobalAttribute = "lang" | "role" | "aria-expanded";
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/globalAttributes.ts:4]: Popover also sets `aria-controls`, so does that also need to be set here? Not sure if we need `aria-describedby` to support tooltip. cc @geospatialem

**Review by jcfranco — APPROVED** [review]
🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸
🎸🤘🎸🎸🎸🤘🎸🤘🎸🎸🎸🤘🎸🎸🤘🤘🤘🎸🤘🎸
🎸🎸🤘🎸🤘🎸🎸🤘🎸🎸🎸🤘🎸🤘🎸🎸🎸🎸🤘🎸
🎸🎸🎸🤘🎸🎸🎸🤘🎸🎸🎸🤘🎸🎸🤘🤘🎸🎸🤘🎸
🎸🎸🎸🤘🎸🎸🎸🤘🎸🎸🎸🤘🎸🎸🎸🎸🤘🎸🎸🎸
🎸🎸🎸🤘🎸🎸🎸🎸🤘🤘🤘🎸🎸🤘🤘🤘🎸🎸🤘🎸
🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸🎸

File: packages/calcite-components/src/components/button/button.e2e.ts (hunk 1 above)
Comment (anveshmekala) [reply packages/calcite-components/src/components/button/button.e2e.ts:648]: `accessible` helper is unable to pick this one up.

File: packages/calcite-components/src/utils/globalAttributes.ts (hunk 2 above)
Comment (anveshmekala) [reply packages/calcite-components/src/utils/globalAttributes.ts:4]: FWIW, `aria-controls` need not be parsed on to the native button for the following reasons:
-It has minimal support across screen readers in terms of providing context as per https://a11ysupport.io/tech/aria/aria-controls_attribute. 
- For scenarios where screen reader should jump directly onto the opened element, the popover is currently doing that with or without aria-controls.

**Review by geospatialem — COMMENTED** [review]
This will be a fantastic fix! 🌟 

Observed the `button` doesn't appear to be selectable/clickable via the `Enter` key when testing with NVDA and Firefox, but not sure what could be causing the issue. 

Here's the code tested with (taken from the original issue Codepen):
```html
<calcite-label layout="inline"
  ><span>Age groups:</span>
  <calcite-button id="test-category-filter-button" icon-end="chevron-down" width="half" label="Any age groups">
    Any age groups
  </calcite-button>
  <calcite-popover
    label="Select types"
    reference-element="test-category-filter-button"
    auto-close
    placement="bottom"
    id="category-filter-popover"
  >
    <p style="padding: 12px; margin: 0">
      <calcite-label layout="inline">
        <calcite-checkbox value="infant"></calcite-checkbox>
        infant
      </calcite-label>
      <calcite-label layout="inline">
        <calcite-checkbox value="toddler"></calcite-checkbox>
        toddler
      </calcite-label>
      <calcite-label layout="inline">
        <calcite-checkbox value="preschool"></calcite-checkbox>
        preschool
      </calcite-label>
      <calcite-label layout="inline">
        <calcite-checkbox value="school age"></calcite-checkbox>
        school age
      </calcite-label>
    </p>
  </calcite-popover>
</calcite-label>
```

File: packages/calcite-components/src/utils/globalAttributes.ts (hunk 2 above)
Comment (anveshmekala) [reply packages/calcite-components/src/utils/globalAttributes.ts:4]: In VoiceOver, context is provided when user check/uncheck the checkbox before closing the popup.

**Comment by anveshmekala** [comment]
> This will be a fantastic fix! 🌟
> 
> Observed the `button` doesn't appear to be selectable/clickable via the `Enter` key when testing with NVDA and Firefox, but not sure what could be causing the issue.
> 
 Is it possible that NVDA is in browse mode? Browse mode can intercept certain keys from working as expected.

**Comment by geospatialem** [comment]
> > Observed the `button` doesn't appear to be selectable/clickable via the `Enter` key when testing with NVDA and Firefox, but not sure what could be causing the issue.
> 
> Is it possible that NVDA is in browse mode? Browse mode can intercept certain keys from working as expected.


That was the trick! ✨ Only occurs in Firefox, and was discovered as a similar report (which was closed) in https://github.com/nvaccess/nvda/issues/4903. More info: https://dequeuniversity.com/screenreaders/nvda-keyboard-shortcuts#nvda-browse_focus_modes

File: packages/calcite-components/src/utils/globalAttributes.ts (hunk 2 above)
Comment (geospatialem) [reply packages/calcite-components/src/utils/globalAttributes.ts:4]: Windows OS + JAWS updates this week mitigated the issue from last week. 💪🏻

**Review by geospatialem — APPROVED** [review]
LGTM 🚀

**Pull request merged**